checkST2110.js

```diff
@@ -1073,108 +1305,317 @@ const test_21_82_2 = (sdp, params) => {
+  let errors = [];
+  let lines = splitLines(sdp);
+  let bandwidthPresent = false;
+  for (let x = 0; x < lines.length; x++) {
+    if (lines[x].startsWith('b=')) {
+      let bandwidthMatch = lines[x].match(bandwidthPattern);
+      if (bandwidthMatch == null) {
+        errors.push(new Error(`Line ${x + 1}: In 'Bandwidth must be of the form 'b=<bwtype>:<bandwidth>' as per SMPTE ST2110-22 Section 7.3.`));
+        continue;
+      }
+      if (bandwidthMatch[1] != 'AS')
+        errors.push(new Error(`Line ${x + 1}: In 'b=<bwtype>:<bandwidth>' bwtype must be 'AS' as per SMPTE ST2110-22 Section 7.3.`));
+      if (Number.isInteger(+bandwidthMatch[2]) == false)
+        errors.push(new Error(`Line ${x + 1}}: In 'b=<bwtype>:<bandwidth>' bandwidth must be specified as an integer as per SMPTE ST2110-22 Section 7.3.`));
+      bandwidthPresent = true;
+    }
+  }
+  if (!bandwidthPresent) {
+    errors.push(new Error('Required attribute \'b=<bwtype>:<bandwidth>\' is missing, as per SMPTE ST2110-22 Section 7.3.'));
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: Test ST 2110-22 Section 7.3 -  Check for mandatory bitrate attribute');
   }
   return errors;
 };
 
+// Test ST 2110-22 Section 74 Test 1-  Check that framerate is specified by one of accepted methods
+const test_22_74_1 = (sdp, params) => {
+  let lines = splitLines(sdp);
+  let framerateAttributePresent = false;
+  let framerateParameterPresent = false;
+  // First check if exactframerate is specified as a parameter of fmtp
+  let [mtParams, errors] = extractMTParams(sdp, params);
+  if(mtParams.length == 1) {
+    if(mtParams[0].exactframerate != null) {
+      framerateParameterPresent = true;
+    }
+  }else {
+    errors.push(new Error('SMPTE-2110-22 SDP File should specify a single video stream or a duplicate for SMPTE2110-7'));
+  }
+  // Now check if it's present as an attribute
+  for (let x = 0; x < lines.length; x++) {
+    if (lines[x].startsWith('a=framerate')) {
+      let framerateMatch = lines[x].match(frameRateAttributePattern);
+      if (framerateMatch == null) {
+        errors.push(new Error(`Line ${x + 1}: In 'framerate must be of the form a=framerate:<frame rate> (no trailing '.' or '0's ) as per SMPTE ST2110-22 Section 7.3.`));
+        continue;
+      }
+      framerateAttributePresent = true;
+    }
+  }
+  // If neither exactframerate or framerate attribute present - load up error 
+  if (!framerateAttributePresent && !framerateParameterPresent) {
+    errors.push(new Error('framerate must specified as either an attribute or a parmameter of video fmtp as per SMPTE ST2110-22 Section 7.4.'));
+  }
+  //  If both specified then error.  SMPTE-2110-22 section 7.4 indicates one method of specifying
+  if (framerateAttributePresent && framerateParameterPresent) {
+    errors.push(new Error('framerate must be specified using one method only (attribute or a parmameter of video fmtp) as per SMPTE ST2110-22 Section 7.4.'));
```

Comment:
Typo "parmameter" in a couple of places.

Reply:
corrected and closing the item.